### PR TITLE
Modifications to allow using network ELB type

### PR DESCRIPTION
**Pull request by craigharman**
I have conditionally modified elb_settings to allow for ELB type of "network" as it was not working previously. 

This is required as:

- Protocol needs to be TCP not HTTP
- No health check
- S3 logs and Security Groups are not valid for Network ELB

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -464,28 +454,45 @@ locals {
...
+      value     = var.loadbalancer_certificate_arn
+    },
+    {
+      namespace = "aws:elbv2:loadbalancer"
+      name      = "SecurityGroups"
+      value     = join(",", var.loadbalancer_security_groups)
     }
   ]
 
+  elb_settings = local.application_settings == null ? local.initial_elb_settings : "${merge(
```
Comment (aknysh) [inline main.tf]: why use interpolation 0.11 syntax `"${merge`?
can it just be `merge(local.initial_elb_settings, local.application_settings)`?

**Comment by craigharman** [comment]
No problems @aknysh I have implemented those requested changes and run the make commands.

**Comment by craigharman** [comment]
I've added the variable but when I run make readme I get a build error and not familiar with what this doing?

curl --retry 3 --retry-delay 5 --fail -sSL -o /Users/craig/Code/terraform-aws-elastic-beanstalk-environment/build-harness/vendor/terraform-docs https://github.com/segmentio/terraform-docs/releases/download/v0.4.5/terraform-docs-v0.4.5-darwin-amd64 && chmod +x <PATHREMOVED>/terraform-aws-elastic-beanstalk-environment/build-harness/vendor/terraform-docs
make: gomplate: No such file or directory
make: *** [readme/build] Error 1

**Comment by Gowiem** [comment]
@craigharman Mind addressing the conflicts and requests above?

**Comment by florian0410** [comment]
Hello @craigharman is this PR still under work on your side ? I would be glad to have this feature for this module.

**Comment by craigharman** [comment]
I haven’t worked on this for ages I’m afraid.

—
Craig Harman

**Pull request closed without merging**